
Co-authored by Nali Toja

Index: kfreebsd-10/sys/conf/kern.mk
===================================================================
--- kfreebsd-10.orig/sys/conf/kern.mk
+++ kfreebsd-10/sys/conf/kern.mk
@@ -1,5 +1,8 @@
 # $FreeBSD$
 
+# Debian compilers do not have this FreeBSD-specific compiler extension
+MK_FORMAT_EXTENSIONS=no
+
 #
 # Warning flags for compiling the kernel and components of the kernel:
 #
@@ -67,7 +70,7 @@ FORMAT_EXTENSIONS=	-fformat-extensions
 #
 .if ${MACHINE_CPUARCH} == "i386"
 .if ${COMPILER_TYPE} != "clang"
-CFLAGS+=	-mno-align-long-strings -mpreferred-stack-boundary=2
+CFLAGS+=	${no-align-long-strings} -mpreferred-stack-boundary=2
 .else
 CFLAGS+=	-mno-aes -mno-avx
 .endif
