
Co-authored by Nali Toja

--- a/sys/conf/kern.mk
+++ b/sys/conf/kern.mk
@@ -1,5 +1,8 @@
 # $FreeBSD: stable/10/sys/conf/kern.mk 268882 2014-07-19 18:33:09Z dim $
 
+# Debian compilers do not have this FreeBSD-specific compiler extension
+MK_FORMAT_EXTENSIONS=no
+
 #
 # Warning flags for compiling the kernel and components of the kernel:
 #
@@ -62,7 +65,7 @@
 #
 .if ${MACHINE_CPUARCH} == "i386"
 .if ${COMPILER_TYPE} != "clang"
-CFLAGS+=	-mno-align-long-strings -mpreferred-stack-boundary=2
+CFLAGS+=	${no-align-long-strings} -mpreferred-stack-boundary=2
 .else
 CFLAGS+=	-mno-aes -mno-avx
 .endif
