
Merged (rev 261497).

Index: kfreebsd-10/sys/dev/drm2/radeon/r600.c
===================================================================
--- kfreebsd-10.orig/sys/dev/drm2/radeon/r600.c
+++ kfreebsd-10/sys/dev/drm2/radeon/r600.c
@@ -3024,6 +3024,12 @@ int r600_init(struct radeon_device *rdev
 		return -EINVAL;
 	}
 
+	/* Don't start up if the ucode is missing. */
+	if (!rdev->me_fw || !rdev->pfp_fw || !rdev->rlc_fw) {
+		DRM_ERROR("radeon: ucode required for R600+.\n");
+		return -EINVAL;
+	}
+
 	return 0;
 }
 
Index: kfreebsd-10/sys/dev/drm2/radeon/rv770.c
===================================================================
--- kfreebsd-10.orig/sys/dev/drm2/radeon/rv770.c
+++ kfreebsd-10/sys/dev/drm2/radeon/rv770.c
@@ -1195,6 +1195,12 @@ int rv770_init(struct radeon_device *rde
 		return -EINVAL;
 	}
 
+	/* Don't start up if the ucode is missing. */
+	if (!rdev->me_fw || !rdev->pfp_fw || !rdev->rlc_fw) {
+		DRM_ERROR("radeon: ucode required for R600+.\n");
+		return -EINVAL;
+	}
+
 	return 0;
 }
 
